modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/PartitionAccess.java

```diff
@@ -17,43 +17,44 @@
 
 package org.apache.ignite.internal.table.distributed.raft.snapshot;
 
-import java.util.List;
-import org.apache.ignite.internal.storage.ReadResult;
-import org.apache.ignite.internal.storage.RowId;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import org.apache.ignite.internal.storage.MvPartitionStorage;
 import org.apache.ignite.internal.storage.StorageException;
-import org.jetbrains.annotations.Nullable;
+import org.apache.ignite.internal.tx.storage.state.TxStateStorage;
 
 /**
  * Small abstractions for partition storages that includes only methods, mandatory for the snapshot storage.
  */
 public interface PartitionAccess {
     /**
      * Returns the key that uniquely identifies the corresponding partition.
-     *
-     * @return Partition key.
      */
-    PartitionKey key();
+    PartitionKey partitionKey();
+
+    /**
+     * Returns the multi-versioned partition storage.
+     */
+    MvPartitionStorage mvPartitionStorage();
 
     /**
-     * Returns persisted RAFT index for the partition.
+     * Returns transaction state storage for the partition.
      */
-    long persistedIndex();
+    TxStateStorage txStatePartitionStorage();
 
     /**
-     * Returns a row id, existing in the storage, that's greater or equal than the lower bound. {@code null} if not found.
+     * Destroys and recreates the multi-versioned partition storage.
      *
-     * @param lowerBound Lower bound.
-     * @throws StorageException If failed to read data from the storage.
+     * @param executor Executor.
+     * @throws StorageException If an error has occurred during the partition destruction.
      */
-    @Nullable
-    RowId closestRowId(RowId lowerBound);
+    CompletableFuture<MvPartitionStorage> reCreateMvPartitionStorage(Executor executor) throws StorageException;
```

Comment:
Are you suggesting calling **TxStateStorage#destroy** instead of **TxStateTableStorage#destroyTxStateStorage**?
I think it's better to call **TxStateTableStorage#destroyTxStateStorage**. And don't show **TxStateTableStorage** to the outside.

Wdyt?

Reply:
I forgot that destroy is called from the table and not from the partition, so we can leave it, ignore my comment then

Reply:
Glory to programming!